main/tasks/stratum_task.c

```diff
@@ -222,80 +188,57 @@ void stratum_task(void * pvParameters)
+        is_tls = GLOBAL_STATE->SYSTEM_MODULE.is_using_fallback ? GLOBAL_STATE->SYSTEM_MODULE.fallback_pool_is_tls : GLOBAL_STATE->SYSTEM_MODULE.pool_is_tls;
+        cert = GLOBAL_STATE->SYSTEM_MODULE.is_using_fallback ? GLOBAL_STATE->SYSTEM_MODULE.fallback_pool_cert : GLOBAL_STATE->SYSTEM_MODULE.pool_cert;
+        retry_critical_attempts = 0;
 
-        struct hostent *dns_addr = gethostbyname(stratum_url);
-        if (dns_addr == NULL) {
-            retry_attempts++;
-            vTaskDelay(1000 / portTICK_PERIOD_MS);
-            continue;
-        }
-        inet_ntop(AF_INET, (void *)dns_addr->h_addr_list[0], host_ip, sizeof(host_ip));
-
-        ESP_LOGI(TAG, "Connecting to: stratum+tcp://%s:%d (%s)", stratum_url, port, host_ip);
-
-        struct sockaddr_in dest_addr;
-        dest_addr.sin_addr.s_addr = inet_addr(host_ip);
-        dest_addr.sin_family = AF_INET;
-        dest_addr.sin_port = htons(port);
-
-        GLOBAL_STATE->sock = socket(addr_family, SOCK_STREAM, ip_protocol);
-        if (GLOBAL_STATE->sock < 0) {
-            ESP_LOGE(TAG, "Unable to create socket: errno %d", errno);
+        GLOBAL_STATE->transport = STRATUM_V1_transport_init(is_tls, cert);
+        // Check if transport was initialized
+        if(GLOBAL_STATE->transport == NULL) {
+            ESP_LOGE(TAG, "Transport initialization failed.");
             if (++retry_critical_attempts > MAX_CRITICAL_RETRY_ATTEMPTS) {
                 ESP_LOGE(TAG, "Max retry attempts reached, restarting...");
                 esp_restart();
             }
             vTaskDelay(5000 / portTICK_PERIOD_MS);
             continue;
         }
-        retry_critical_attempts = 0;
-
-        ESP_LOGI(TAG, "Socket created, connecting to %s:%d", host_ip, port);
-        int err = connect(GLOBAL_STATE->sock, (struct sockaddr *)&dest_addr, sizeof(struct sockaddr_in6));
-        if (err != 0)
-        {
-            retry_attempts++;
-            ESP_LOGE(TAG, "Socket unable to connect to %s:%d (errno %d: %s)", stratum_url, port, errno, strerror(errno));
-            // close the socket
-            shutdown(GLOBAL_STATE->sock, SHUT_RDWR);
-            close(GLOBAL_STATE->sock);
+        ESP_LOGI(TAG, "Transport initialized, connecting to %s:%d", stratum_url, port);
+        esp_err_t ret = STRATUM_V1_transport_connect(stratum_url, port, GLOBAL_STATE->transport);
+        if (ret != ESP_OK) {
+            retry_attempts ++;
+            ESP_LOGE(TAG, "Transport unable to connect to %s:%d (errno %d). Attempt: %d", stratum_url, port, ret, retry_attempts);
+            // close the transport
+            STRATUM_V1_transport_close(GLOBAL_STATE->transport);
             // instead of restarting, retry this every 5 seconds
             vTaskDelay(5000 / portTICK_PERIOD_MS);
             continue;
         }
 
-        if (setsockopt(GLOBAL_STATE->sock, SOL_SOCKET, SO_SNDTIMEO, &tcp_snd_timeout, sizeof(tcp_snd_timeout)) != 0) {
```

Comment:
Where is this defined now?

Reply:
I have replaced all `sock` with `tcp_transport`.